tests/general/KeywordBearTest.py

```diff
@@ -14,3 +22,170 @@
+        with execute_bear(self.uut, "F", text,
+                          dependency_results=dep_results) as result:
+            self.assertEqual(result[0].diffs, {})
+            self.assertEqual(result[0].affected_code[0].start.line, 1)
+            self.assertEqual(len(result), 1)
+
+        dep_results = {'AnnotationBear': HiddenResult(
+            'AnnotationBear', 123)}
+        with execute_bear(self.uut, "F", text,
+                          dependency_results=dep_results) as result:
+            self.assertEqual(result[0].diffs, {})
+            self.assertEqual(result[0].affected_code[0].start.line, 1)
+            self.assertEqual(len(result), 1)
+
+        text = ['# comment 123\n', 'a = "TODO"\n']
+        comments = [SourceRange.from_values("F", 1, 1, 1, 40)]
+        dep_results = {
+            'AnnotationBear': [
+                self.annotation_bear_result_type({'comments': comments})
+            ]
+        }
+
+        with execute_bear(self.uut, "F", text,
+                          dependency_results=dep_results) as result:
+            self.assertEqual(len(result[0].diffs), 0)
+
+        text = ['# todo 123\n']
+        comments = [SourceRange.from_values("F", 1, 1, 1, 10)]
+        dep_results = {
+            'AnnotationBear': [
+                self.annotation_bear_result_type({'comments': comments})
+            ]
+        }
+
+        with execute_bear(self.uut, "F", text,
+                          dependency_results=dep_results) as result:
+            self.assertEqual(result[0].diffs['F'].unified_diff,
+                             '--- \n'
+                             '+++ \n'
+                             '@@ -1 +0,0 @@\n'
+                             '-# todo 123\n')
+
+        text = ['test = 55 # todo 123\n']
+        comments = [SourceRange.from_values("F", 1, 11, 1, 23)]
+        dep_results = {
+            'AnnotationBear': [
+                self.annotation_bear_result_type({'comments': comments})
+            ]
+        }
+        with execute_bear(self.uut, "F", text,
+                          dependency_results=dep_results) as result:
+            self.assertEqual(result[0].diffs['F'].unified_diff,
+                             '--- \n'
+                             '+++ \n'
+                             '@@ -1 +1 @@\n'
+                             '-test = 55 # todo 123\n'
+                             '+test = 55\n')
+
+    def test_keyword_outside_of_comment(self):
+        section = Section("")
+        section.append(Setting('language', 'python3'))
+        section.append(Setting('keywords', 'todo'))
+        uut = KeywordBear(section, Queue())
+
+        text = ['todo = 123\n']
+        with execute_bear(uut, "F", text,
+                          dependency_results=self.dep_results) as result:
+            self.assertEquals(result[0].diffs, {})
+
+    def test_keyword_between_code(self):
+        section = Section("")
+        section.append(Setting('language', 'c'))
```

Comment:
same here, use self.section and override the language setting instead of making a new one